### PR TITLE
[metallb] Distroless

### PR DESCRIPTION
**Pull request by vitaliy-sn**
## Description

* The MetalLB docker images are now distroless.
* MetalLB was updated to v0.13.11.
    *  New version requires access (`get`, `list`, `watch`) to `configmaps` and `namespaces`.

## Tests

<details>

```shell
v.snurnitsin@thinkpad:/tmp/tmp$ docker save dev-registry.deckhouse.io/sys/deckhouse-oss@sha256:0053f901be3a6c076eeaabd55bb9cf2ac9833eaa1be6c7da63d0070ba090339d > 1.tar
v.snurnitsin@thinkpad:/tmp/tmp$ tar xvf 1.tar 
30526b51275d9a1659f5348b7aa34b0c228d15edfd91ab97560cd73cfeced9ac/
30526b51275d9a1659f5348b7aa34b0c228d15edfd91ab97560cd73cfeced9ac/VERSION
30526b51275d9a1659f5348b7aa34b0c228d15edfd91ab97560cd73cfeced9ac/json
30526b51275d9a1659f5348b7aa34b0c228d15edfd91ab97560cd73cfeced9ac/layer.tar
4fbd3b757ac5297cf966dba17c269219f9ef03d8c7f4ea1400f8fe2348a2bb71.json
baef073e354c4355169c716d86bc41490550d7329ffbeb3f61282f662c6cfa4f/
baef073e354c4355169c716d86bc41490550d7329ffbeb3f61282f662c6cfa4f/VERSION
baef073e354c4355169c716d86bc41490550d7329ffbeb3f61282f662c6cfa4f/json
baef073e354c4355169c716d86bc41490550d7329ffbeb3f61282f662c6cfa4f/layer.tar
e2478ebaad1193040a8531a3103250a8dddcd59d021639ced565a49756542361/
e2478ebaad1193040a8531a3103250a8dddcd59d021639ced565a49756542361/VERSION
e2478ebaad1193040a8531a3103250a8dddcd59d021639ced565a49756542361/json
e2478ebaad1193040a8531a3103250a8dddcd59d021639ced565a49756542361/layer.tar
manifest.json
v.snurnitsin@thinkpad:/tmp/tmp$ tree -h
[4.0K]  .
├── [ 40M]  1.tar
├── [4.0K]  30526b51275d9a1659f5348b7aa34b0c228d15edfd91ab97560cd73cfeced9ac
│   ├── [ 482]  json
│   ├── [2.5K]  layer.tar
│   └── [   3]  VERSION
├── [5.5K]  4fbd3b757ac5297cf966dba17c269219f9ef03d8c7f4ea1400f8fe2348a2bb71.json
├── [4.0K]  baef073e354c4355169c716d86bc41490550d7329ffbeb3f61282f662c6cfa4f
│   ├── [ 406]  json
│   ├── [3.3M]  layer.tar
│   └── [   3]  VERSION
├── [4.0K]  e2478ebaad1193040a8531a3103250a8dddcd59d021639ced565a49756542361
│   ├── [2.6K]  json
│   ├── [ 37M]  layer.tar
│   └── [   3]  VERSION
└── [ 343]  manifest.json

4 directories, 12 files
v.snurnitsin@thinkpad:/tmp/tmp$ cd e2478ebaad1193040a8531a3103250a8dddcd59d021639ced565a49756542361
v.snurnitsin@thinkpad:/tmp/tmp/e2478ebaad1193040a8531a3103250a8dddcd59d021639ced565a49756542361$ tar xvf layer.tar 
controller
v.snurnitsin@thinkpad:/tmp/tmp/e2478ebaad1193040a8531a3103250a8dddcd59d021639ced565a49756542361$ ldd controller 
        not a dynamic executable
v.snurnitsin@thinkpad:/tmp/tmp/e2478ebaad1193040a8531a3103250a8dddcd59d021639ced565a49756542361$
```

```shell
v.snurnitsin@thinkpad:/tmp/tmp$ docker save dev-registry.deckhouse.io/sys/deckhouse-oss@sha256:451e90077a87964efe49274e0a5b7d320f490809fb2dfb91840a7a9a80a148b2 > 1.tar
v.snurnitsin@thinkpad:/tmp/tmp$ tar xvf 1.tar 
30526b51275d9a1659f5348b7aa34b0c228d15edfd91ab97560cd73cfeced9ac/
30526b51275d9a1659f5348b7aa34b0c228d15edfd91ab97560cd73cfeced9ac/VERSION
30526b51275d9a1659f5348b7aa34b0c228d15edfd91ab97560cd73cfeced9ac/json
30526b51275d9a1659f5348b7aa34b0c228d15edfd91ab97560cd73cfeced9ac/layer.tar
683418cac7b5616c4e7458177c7cc819c52003d827ea2137ab140b594fbca02a/
683418cac7b5616c4e7458177c7cc819c52003d827ea2137ab140b594fbca02a/VERSION
683418cac7b5616c4e7458177c7cc819c52003d827ea2137ab140b594fbca02a/json
683418cac7b5616c4e7458177c7cc819c52003d827ea2137ab140b594fbca02a/layer.tar
baef073e354c4355169c716d86bc41490550d7329ffbeb3f61282f662c6cfa4f/
baef073e354c4355169c716d86bc41490550d7329ffbeb3f61282f662c6cfa4f/VERSION
baef073e354c4355169c716d86bc41490550d7329ffbeb3f61282f662c6cfa4f/json
baef073e354c4355169c716d86bc41490550d7329ffbeb3f61282f662c6cfa4f/layer.tar
c0fda5a9e3abf71ee88e16b8972206ffd78a7f83d73221fd243fc48d5dc5758b.json
manifest.json
v.snurnitsin@thinkpad:/tmp/tmp$ tree -h
[4.0K]  .
├── [ 43M]  1.tar
├── [4.0K]  30526b51275d9a1659f5348b7aa34b0c228d15edfd91ab97560cd73cfeced9ac
│   ├── [ 482]  json
│   ├── [2.5K]  layer.tar
│   └── [   3]  VERSION
├── [4.0K]  683418cac7b5616c4e7458177c7cc819c52003d827ea2137ab140b594fbca02a
│   ├── [2.6K]  json
│   ├── [ 40M]  layer.tar
│   └── [   3]  VERSION
├── [4.0K]  baef073e354c4355169c716d86bc41490550d7329ffbeb3f61282f662c6cfa4f
│   ├── [ 406]  json
│   ├── [3.3M]  layer.tar
│   └── [   3]  VERSION
├── [5.5K]  c0fda5a9e3abf71ee88e16b8972206ffd78a7f83d73221fd243fc48d5dc5758b.json
└── [ 343]  manifest.json

4 directories, 12 files
v.snurnitsin@thinkpad:/tmp/tmp$ cd 683418cac7b5616c4e7458177c7cc819c52003d827ea2137ab140b594fbca02a
v.snurnitsin@thinkpad:/tmp/tmp/683418cac7b5616c4e7458177c7cc819c52003d827ea2137ab140b594fbca02a$ tar xvf layer.tar 
speaker
v.snurnitsin@thinkpad:/tmp/tmp/683418cac7b5616c4e7458177c7cc819c52003d827ea2137ab140b594fbca02a$ ldd speaker 
        not a dynamic executable
v.snurnitsin@thinkpad:/tmp/tmp/683418cac7b5616c4e7458177c7cc819c52003d827ea2137ab140b594fbca02a$
```

```shell
root@s-metallb-test-ks-master-0:~# kubectl -n d8-metallb get po -o wide
NAME                          READY   STATUS    RESTARTS       AGE    IP              NODE                                                  NOMINATED NODE   READINESS GATES
controller-7bfcdf4bb6-48kd2   2/2     Running   3 (104m ago)   110m   10.111.2.139    s-metallb-test-ks-system-e2b910ae-679bc-njbtl         <none>           <none>
speaker-pg29z                 2/2     Running   9 (75m ago)    110m   172.16.28.239   s-metallb-test-ks-loadbalancer-e2b910ae-84665-xkxmt   <none>           <none>
root@s-metallb-test-ks-master-0:~# kubectl -n d8-metallb logs controller-7bfcdf4bb6-48kd2 controller | grep version\ 0
{"branch":"","caller":"main.go:155","commit":"","goversion":"gc / go1.19.3 / amd64","level":"info","msg":"MetalLB controller starting version 0.13.11 (no build information)","ts":"2023-10-15T18:05:54Z","version":"0.13.11"}
root@s-metallb-test-ks-master-0:~# kubectl -n d8-metallb logs speaker-pg29z speaker | grep version\ 0
{"branch":"","caller":"main.go:100","commit":"","goversion":"gc / go1.19.3 / amd64","level":"info","msg":"MetalLB speaker starting version 0.13.11 (no build information)","ts":"2023-10-15T18:38:58Z","version":"0.13.11"}
root@s-metallb-test-ks-master-0:~# kubectl -n d8-metallb logs speaker-pg29z speaker | grep -i error
root@s-metallb-test-ks-master-0:~# kubectl -n d8-metallb logs controller-7bfcdf4bb6-48kd2 controller | grep -i error
root@s-metallb-test-ks-master-0:~# kubectl -n d8-ingress-nginx get svc | grep LoadBalancer
main-load-balancer       LoadBalancer   10.222.48.244    10.220.212.16   80:31602/TCP,443:32159/TCP   2d9h
root@s-metallb-test-ks-master-0:~# kubectl -n d8-metallb logs speaker-pg29z speaker | grep 10.220.212.16
{"caller":"bgp_controller.go:309","event":"updatedAdvertisements","ips":["10.220.212.16"],"level":"info","msg":"making advertisements using BGP","numAds":1,"pool":"mypool","protocol":"bgp","ts":"2023-10-15T18:39:20Z"}
{"caller":"main.go:374","event":"serviceAnnounced","ips":["10.220.212.16"],"level":"info","msg":"service has IP, announcing","pool":"mypool","protocol":"bgp","ts":"2023-10-15T18:39:20Z"}
root@s-metallb-test-ks-master-0:~#
```

</details>

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: metallb
type: feature
summary: The metallb docker images are now distroless
impact: The metallb pods will be restarted
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/aws'...

File: ee/modules/380-metallb/images/controller/werf.inc.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+artifact: {{ .ModuleName }}/controller-artifact
+from: {{ .Images.BASE_GOLANG_19_ALPINE }}
+shell:
+  install:
+    - apk add --no-cache git
+    - mkdir -p /src
+    - cd /src
+    - git clone --depth 1 -b v0.13.11 https://github.com/metallb/metallb.git .
```
Comment (RomanenkoDenys) [inline ee/modules/380-metallb/images/controller/werf.inc.yaml:9]: SOURCE_REPO and GOPROXY

File: ee/modules/380-metallb/images/speaker/werf.inc.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+---
+artifact: {{ .ModuleName }}/speaker-artifact
+from: {{ .Images.BASE_GOLANG_19_ALPINE }}
+shell:
+  install:
+    - apk add --no-cache git
+    - mkdir -p /src
+    - cd /src
+    - git clone --depth 1 -b v0.13.11 https://github.com/metallb/metallb.git .
```
Comment (RomanenkoDenys) [inline ee/modules/380-metallb/images/speaker/werf.inc.yaml:9]: same

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
aslo i think we should use one artifact to build all binaries

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/aws'...

File: ee/modules/380-metallb/images/speaker/werf.inc.yaml (hunk 2 above)
Comment (vitaliy-sn) [reply ee/modules/380-metallb/images/speaker/werf.inc.yaml]: done

File: ee/modules/380-metallb/images/controller/werf.inc.yaml (hunk 1 above)
Comment (vitaliy-sn) [reply ee/modules/380-metallb/images/controller/werf.inc.yaml]: done

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/aws'...

**Review by name212 — APPROVED** [review]

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/aws'...

**Review by apolovov — APPROVED** [review]

**Pull request merged**